lib/v2/blocs/rates/mappers/rates_state_mapper.dart

```diff
@@ -0,0 +1,23 @@
+import 'package:seeds/v2/datasource/remote/model/fiat_rate_model.dart';
+import 'package:seeds/v2/datasource/remote/model/rate_model.dart';
+import 'package:seeds/v2/domain-shared/page_state.dart';
+import 'package:seeds/v2/domain-shared/result_to_state_mapper.dart';
+import 'package:seeds/v2/blocs/rates/viewmodels/rates_state.dart';
+
+class RatesStateMapper extends StateMapper {
+  RatesState mapResultToState(RatesState currentState, List<Result> results) {
+    if (areAllResultsError(results)) {
+      return currentState.copyWith(pageState: PageState.failure, errorMessage: 'Cannot fetch balance...');
+    } else {
+      print('RatesStateMapper mapResultsToState length=${results.length}');
+      results.retainWhere((Result i) => i.isValue);
+      var values = results.map((Result i) => i.asValue.value).toList();
+      RateModel rate = values.firstWhere((i) => i is RateModel, orElse: () => null);
+      List<FiatRateModel> fiatRates = values.whereType<FiatRateModel>().toList();
+      var fiatRate = fiatRates[0];
+      fiatRate.merge(fiatRates[1]);
```

Comment:
how can we be sure this is not going to throw a out of bound exception ?

Reply:
ok i will do tests

Reply:
What do you think about the solution I did, although I think it would be an edget case that the two end-points do not return a response.